virttest/qemu_devices/qcontainer.py

```diff
@@ -2548,9 +2549,18 @@ def dimm_device_define_by_params(self, params, name):
         params = params.object_params("dimm")
         dimm_type = "nvdimm" if params.get("nv_backend") else "pc-dimm"
         attrs = qdevices.Dimm.__attributes__[:]
-        params = params.copy_from_keys(attrs)
-        dev = qdevices.Dimm(params=params, dimm_type=dimm_type)
+        dev = qdevices.Dimm(params=params.copy_from_keys(attrs),
+                            dimm_type=dimm_type)
         dev.set_param("id", "%s-%s" % ("dimm", name))
+        if dimm_type == "nvdimm" and params.get("nvdimm_uuid"):
+            try:
+                dev.set_param("uuid", uuid.UUID(params["nvdimm_uuid"]))
+            except ValueError as err:
+                logging.warning("%s, will generate a random UUID base on uuid5"
+                                " and '%s'", err, name)
+                dev.set_param("uuid", uuid.uuid5(uuid.NAMESPACE_OID, name))
```

Comment:
I'd suggest calling this `<auto>` because it not actually generate "random" hexadecimal digits when using some version (uuid5), and simply setting the option with the literal string user provides (which means we allow wrong values and let qemu complain about), then the code would be like:
```python
nvdimm_uuid = params.get("nvdimm_uuid")
if nvdimm_uuid == "<auto>":
    nvdimm_uuid = uuid.uuid5(uuid.NAMESPACE_OID, name)
dev.set_param("uuid", nvdimm_uuid)
```

Reply:
Okay, I got your point that if we config a wrong format of `nvdimm_uuid`, let vt forces set it to assign a negative value, and raise error when create_qemu_command.